packages/headless/src/auth/index.html

```diff
@@ -0,0 +1,37 @@
+<html>
+  <head>
+    <script type="module">
+      import {buildSamlClient} from './saml-client';
+
+      const organizationId = 'pvnh4ja7n5y7lmgstt3fpwy2bzq';
+
+      const saml = buildSamlClient({
+        organizationId,
+        provider: 'oktaA323aab78b9f1-45b5-a095-a1f0fa09ddd5',
+      });
+
+      async function main() {
+        const accessToken = await saml.authenticate();
+
+        buildSearchEngine({
+          configuration: {
+            organizationId,
+            accessToken,
+            renewAccessToken: () => saml.authenticate(),
```

Comment:
Just a thought, but we could have a pattern similar to: 

```
const saml = buildSamlClient({orgId, provider})

buildSearchEngine({
  configuration: saml
})

// or
buildSearchEngine({
  configuration: {
      ...saml,
      search: {},
      analytics: {},
   }
})
```

Since it's an opinionated package built explicitly for Headless (that exists as a sub package), it could be neat to couple them tightly together ?

What do you think.

Reply:
I like how the api looks in principle, removing the orgId duplication, and the need to configure `renewAccessToken`. Some considerations though:

- Atm, an `accessToken` (and `organizationId`) is required. I don't think making it optional is a good step as the example suggests. We could a) expand the type of `accessToken` to accept a `SamlClient` or, b) refactor to add an `auth` property which has the benefit of removing the `organizationId` duplication.

```
// option a

buildSearchEngine({
  configuration: {
    organizationId: '...'
    accessToken: saml
  }
})
```
```
// option b
buildSearchEngine({
  auth: saml
})

// where
type Auth = {organizationId: string; accessToken: string} | SamlClient;
```

We can introduce `auth` in a non-breaking way, and break in v2, removing `organizationId` and `accessToken` from `configuration`, logging a warning in the meantime.

- The saml client would need to become subscribable, so that multiple engine instances can listen to the instance and update the access token in their stores. e.g

```
function buildSearchEngine(config) {
   if (config.auth === SamlClient) {
      config.auth.subscribe(token => {
         dispatch(updateAccessToken(token))
      })
    }
}
```

One concern here is it becomes possible to mistakenly create multiple `SamlClient` instances, vs. the current api which reduces the chance of making this mistake (because one has to pass a string for access token).